web-common/src/features/canvas/components/charts/Chart.svelte

```diff
@@ -11,26 +11,28 @@
     V1ComponentSpecRendererProperties,
   } from "@rilldata/web-common/runtime-client";
   import type { View } from "svelte-vega";
-  import { getChartData } from "./selector";
+  import { getChartData, validateChartSchema } from "./selector";
   import type { ChartType } from "./types";
   import { generateSpec, getChartTitle, mergedVlConfig } from "./util";
 
   export let rendererProperties: V1ComponentSpecRendererProperties;
   export let renderer: string;
 
+  const ctx = getCanvasStateManagers();
   const {
     canvasEntity: {
       spec: { getMeasureForMetricView },
     },
-  } = getCanvasStateManagers();
+  } = ctx;
```

Comment:
I'm repetitive in my comments :), but here's another place where Canvas state management could be pulled out